ludwig/automl/automl.py

```diff
@@ -382,7 +382,7 @@ def _model_select(
                     input_feature[ENCODER] = {TYPE: AUTOML_DEFAULT_IMAGE_ENCODER}
 
         # Needs to be outside for loop because merge dict creates deep copy - this prevents image section from setting
-        base_config = merge_dict(base_config, default_configs[TEXT][AUTOML_DEFAULT_TEXT_ENCODER])
+        base_config = merge_dict(base_config, default_configs.get(TEXT, {}).get(AUTOML_DEFAULT_TEXT_ENCODER, {}))
```

Comment:
My sense is that this is old automl code that hasn't been updated since `defaults` have been added. When I edited this code a couple months ago, I was going to try and change it to work with `defaults` but I remember realizing that it would be a more involved change and was pretty focused on schema work and the model editor, so I just changed to make sure it complied with the nesting for encoders and decoders.

Reply:
@tgaddair the default configs dictionary contains default encoder and hyperopt config values for many different model architectures and backbones. The format of the dictionary is something like this:
```
{
	base_config: {
		input_features: [...]
		output_features: [...]
		trainer: [...]
		hyperopt: [...]
	}
	combiner: {
		concat: {
			combiner: {default configuration}
			hyperopt: {default hyperopt search space}
		}
		tabnet: {
			combiner: {default configuration}
			hyperopt: {default hyperopt search space}
		}
		...
	}
	text: {
		bert: {
			hyperopt: {default hyperopt search space for bert}
		}
		...
	}
}
```

However, the top level `text` key will only exist in cases when the dataset has a text feature. However, the `merge_with_defaults` call that was failing happened outside the loop, and outside the condition where we check if a text feature exists. That caused the `KeyError`.